### PR TITLE
fix(#8564): mask more credentials

### PR DESCRIPTION
**Pull request by garethbowen**
# Description

[description]

medic/cht-core#8564

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs

If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8564-mask-more-credentials/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8564-mask-more-credentials/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8564-mask-more-credentials/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by garethbowen** [comment]
I think this fixes it. We may drop haproxy sometime soon, but at least we have e2e testing for it now.

File: tests/e2e/default/logging/logging.wdio-spec.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    const auditLogs = (await collectAuditLogs()).filter(log => log.length);
+    expect(auditLogs.length).to.equal(1);
+    expect(auditLogs[0]).to.contain('name=admin&password=***');
+  });
+
+  it('should mask password basic auth header', async () => {
+    const collectAuditLogs = await utils.collectHaproxyLogs(/POST,\/_replicator/);
+    const body = {
+      user_ctx: { name: 'medic', roles: [ '_admin', '_reader', '_writer' ]},
+      source: { url: 'https://localhost/source', headers: { Authorization: 'Basic bWVkaWM6cGFzc3dvcmQ=' } },
```
[the remark this answers is not in the document]
Comment (dianabarsan) [reply tests/e2e/default/logging/logging.wdio-spec.js:40]: I feel like this should be a pre-commit check somehow.

[the remark this answers is not in the document]
Comment (garethbowen) [reply tests/e2e/default/logging/logging.wdio-spec.js:40]: Ooh good idea. It doesn't look straightforward to set up unfortunately though there is a vscode plugin...

**Review by mrjones-plip — APPROVED** [review]
lgtm!

Before in `4.3.1` and calling `docker logs  four_three_ftw_haproxy_1 -n 1000|grep password`:

```
<150>Sep 20 21:00:14 haproxy[12]: 172.23.0.7,couchdb,200,3,0,0,POST,/_session,-,medic,'{"name":"medic","password":"***"}',375,3,46,'-'
<150>Sep 20 21:02:36 haproxy[12]: 172.23.0.7,couchdb,401,1,0,0,POST,/_session,-,-,'name=medic&password=password',348,1,61,'-'
<150>Sep 20 21:02:37 haproxy[12]: 172.23.0.7,couchdb,200,2,0,0,POST,/_session,-,medic,'name=medic&password=password',383,2,54,'-'
<150>Sep 20 21:02:37 haproxy[12]: 172.23.0.7,couchdb,401,1,0,0,POST,/_session,-,-,'name=medic&password=password',348,1,61,'-'
<150>Sep 20 21:02:37 haproxy[12]: 172.23.0.7,couchdb,200,1,0,0,POST,/_session,-,medic,'name=medic&password=password',383,1,54,'-'
```

After, using builds for `8564-mask-more-credentials` branch and calling `docker logs  mask_creds_8564_haproxy_1 -n 1000|grep password`:

```
<150>Sep 20 21:07:38 haproxy[12]: 172.24.0.7,couchdb,200,3,0,0,POST,/_session,-,medic,'{"name":"medic","password":"***"}',391,3,46,'-'
<150>Sep 20 21:08:34 haproxy[12]: 172.24.0.7,couchdb,200,1,0,0,POST,/_session,-,-,'name=medic&password=***',399,1,54,'-'
<150>Sep 20 21:08:34 haproxy[12]: 172.24.0.7,couchdb,200,1,0,0,POST,/_session,-,-,'name=medic&password=***',399,1,54,'-'
```

Thanks for your excellent testing steps to set up replication!  Also good regex fu so you got the right string here `'name=medic&password=password'` (my password was `password`) :1st_place_medal:

**Pull request merged**